File: code/dart/modular_api/example/modules/greetings/greetings_builder.dart
import 'package:modular_api/modular_api.dart';

import 'usecases/hello_world.dart';

void buildGreetingsModule(ModuleBuilder m) {
  m.usecase(
    'hello-world',
    HelloWorld.fromJson,
    inputExample: HelloWorldInput.example,
    outputExample: HelloWorldOutput.example,
  );
}

File: code/ts/modular_api/example/modules/greetings/greetings_builder.ts
import { ModuleBuilder } from '../../../src/index';
import { HelloWorldInput, HelloWorldOutput, HelloWorld } from './usecases/hello_world';

export function buildGreetingsModule(m: ModuleBuilder): void {
  m.usecase('hello-world', HelloWorld.fromJson, {
    inputClass: HelloWorldInput,
    outputClass: HelloWorldOutput,
  });
}
